minimesos/src/main/java/com/containersol/minimesos/marathon/MarathonContainer.java

```diff
@@ -112,15 +112,41 @@ public void deployApp(String marathonJson) {
             LOGGER.error(msg);
             throw new MinimesosException(msg, e);
         }
-        LOGGER.debug(String.format("Installing an app on marathon %s", getMarathonEndpoint()));
+        LOGGER.debug(String.format("Installing an app on marathon %s", marathonEndpoint));
+    }
+
+    /**
+     * Replaces ${MINIMESOS_[ROLE]}, ${MINIMESOS_[ROLE]_IP} and ${MINIMESOS_[ROLE]_PORT} tokens in the given string with actual values
+     *
+     * @param source string to replace values in
+     * @return updated string
+     */
+    public String replaceTokens(String source) {
+        // received JSON might contain tokens, which should be replaced before the installation
+        List<ClusterProcess> uniqueRoles = ClusterUtil.getDistinctRoleProcesses(getCluster().getMemberProcesses());
+        String updatedJson = source;
+        for (ClusterProcess process : uniqueRoles) {
+            URI serviceUri = process.getServiceUrl();
+            if (serviceUri != null) {
+                updatedJson = replaceToken(updatedJson, "MINIMESOS_" + process.getRole().toUpperCase(), serviceUri.toString());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "MINIMESOS_" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://ci.infra.container-solutions.com/sonar/coding_rules#rule_key=squid%3AS1192)